src/scene/layer-composition.js

```diff
@@ -51,26 +52,21 @@ class LayerComposition extends EventHandler {
         this._meshInstances = [];
         this._meshInstancesSet = new Set();
 
-        // all unique lights from all layers, stored both as an array, and also map for fast search (key is light, values is its index in _lights array)
+        // an array of all unique lights from all layers
         this._lights = [];
+
+        // a map of Light to index in _lights for fast lookup
         this._lightsMap = new Map();
 
-        // each light in _lights has entry here at the same index, storing an array and also a set of shadow casters for it
+        // each entry in _lights has entry of type ShadowCasters here at the same index, storing shadow casters for the light
```

Comment:
wish we had proper typing info :(

Reply:
every day.
But there is stuff we can do to define types for private variables and similar .. but need to wait till we switch our backend code editor to be visual code based so that it can handle it
https://medium.com/@trukrs/type-safe-javascript-with-jsdoc-7a2a63209b76